### PR TITLE
Refine filter modal layout and styling.

### PR DESCRIPTION
**Pull request by mathewjordan**
image

## What does this do?
This adds some design structure our modal layout and cleans breaks out the `<Filter/>` component a bit more, by adding a new `<Modal/>` component that renders active content of the Radix Dialog when `open`.

- Some minimal responsiveness of modal layout is accounted for on `sm` screens.
- A cancel button is now visible in the footer
- The **View Results** submit button will display a count total if it exists
- A string will render in the header if a `q` exists on the searchState, if q is `joan baez`, it will say _"Results for **"joan baez"**_
- This also media query breakpoints as `media.ts` in the styles directory.

## What's not included.
Any styling of the GroupList and Preview components

### CODE REVIEW

File: components/Facets/Filter/Filter.styled.ts [hunk 1]
```diff
@@ -77,40 +79,153 @@ const FilterOverlay = styled(Dialog.Overlay, {
...
+  position: "absolute",
+  bottom: "0",
+  padding: "1rem",
+  backgroundColor: "$black10",
+  width: "100%",
+  boxShadow: "inset 3px 3px 8px #0001",
+  display: "flex",
+  justifyContent: "flex-end",
+
+  button: {
```
Comment (adamjarling) [inline components/Facets/Filter/Filter.styled.ts:119]: Looks great.  Curious how we're going to keep all these styles consistent for common elements, for example button here, or are you not worrying about that yet?

File: components/Facets/Filter/Modal.test.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      expect(button.getAttribute("aria-label")).toBe("Cancel");
+    });
+  });
+
+  it("Has a visible submit button with a total.", () => {
+    renderHelper();
+    const submit = screen.getByTestId("facets-submit");
+    expect(submit).toHaveTextContent("View Results (20)");
+  });
+});
```
Comment (adamjarling) [inline components/Facets/Filter/Modal.test.tsx:46]: Nice test structure here

File: styles/media.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+const media = {
+  xxs: "(max-width: 349px)",
+  xs: "(max-width: 575px)",
+  sm: "(max-width: 767px)",
+  md: "(max-width: 991px)",
+  xl: "(max-width: 1199px)",
+  lg: "(min-width: 1200px)",
+};
+
+export default media;
```
Comment (adamjarling) [inline styles/media.ts:12]: Sweet.

**Review by adamjarling — APPROVED** [review]
Looks good, but see comment about if/should we be thinking about default styles and/or components.  This is on my mind as I'm laying out the Work page.  Should a `h2` be defined somewhere for example, or do we create a `<Headline />` component or something, etc...

File: components/Facets/Filter/Filter.styled.ts (hunk 1 above)
Comment (mathewjordan) [reply components/Facets/Filter/Filter.styled.ts:119]: We should probably start thinking about that very soon, perhaps make a UI directory of some sort that handles common components and their styling for anything not design system.

**Pull request merged**